backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java

```diff
@@ -0,0 +1,355 @@
+      }
+    }
+
+    public Query buildCountQuery() {
+      Query query = session.createQuery(this.countQueryString);
+      setParameterInQuery(query);
+      return query;
+    }
+
+    private String getHQLQueryString() {
+      boolean isNeedOperator = false;
+      String alias = " repo";
+      StringBuilder queryBuilder =
+          new StringBuilder(" FROM ").append(RepositoryEntity.class.getSimpleName()).append(alias);
+
+      StringBuilder joinClause = new StringBuilder();
+      if (this.predicates != null && !this.predicates.isEmpty()) {
+        this.predicates.forEach(
+            keyValueQuery -> {
+              if (keyValueQuery.getKey().contains(ModelDBConstants.LABEL)) {
+                joinClause
+                    .append(" INNER JOIN ")
+                    .append(LabelsMappingEntity.class.getSimpleName())
+                    .append(" ")
+                    .append(ModelDBConstants.LABELS)
+                    .append(" ON ");
+                joinClause
+                    .append(ModelDBConstants.LABELS)
+                    .append(".id.entity_hash = ")
+                    .append(alias)
+                    .append(".")
+                    .append(ModelDBConstants.ID)
+                    .append(" AND ");
+                joinClause
+                    .append(ModelDBConstants.LABELS)
+                    .append(".id.entity_type = ")
+                    .append(IDTypeEnum.IDType.VERSIONING_REPOSITORY.getNumber());
+              }
+            });
+      }
+
+      StringBuilder whereClause = new StringBuilder();
+      if (workspaceDTO != null
+          && workspaceDTO.getWorkspaceId() != null
+          && !workspaceDTO.getWorkspaceId().isEmpty()) {
+        whereClause.append(" WHERE ");
+        whereClause
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.WORKSPACE_ID)
+            .append(" = :")
+            .append(ModelDBConstants.WORKSPACE_ID);
+        parametersMap.put(ModelDBConstants.WORKSPACE_ID, workspaceDTO.getWorkspaceId());
+        whereClause
+            .append(" AND ")
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.WORKSPACE_TYPE)
+            .append(" = :")
+            .append(ModelDBConstants.WORKSPACE_TYPE);
+        parametersMap.put(
+            ModelDBConstants.WORKSPACE_TYPE, workspaceDTO.getWorkspaceType().getNumber());
+        isNeedOperator = true;
+      } else if (this.isNeedToWhereCause) {
+        whereClause.append(" WHERE ");
+      }
+
+      if (this.predicates != null && !this.predicates.isEmpty()) {
+        if (isNeedOperator) {
+          whereClause.append(" AND ");
+        }
+        for (int index = 0; index < this.predicates.size(); index++) {
+          KeyValueQuery keyValueQuery = this.predicates.get(index);
+          if (keyValueQuery.getKey().contains(ModelDBConstants.LABEL)) {
+            whereClause.append(ModelDBConstants.LABELS).append(".id.label ");
```

Comment:
these predicates can not happen on single table. they need to be on different aliases
this is saying `Labels.id.label = "foo" and Labels,id.label = "bar"` which will never be true.
instead, it should be `L1.id.label = "foo" and L2.id.label = "bar"`
unless there is some hibernate magic happening here

if this is unclear create a repo with two tags and construct a  find with keyValuepredicates on both the tags

Reply:
Fixed logic